### DISCUSSION TITLE
Fill slot without React

### DISCUSSION THREAD

**Discussion by multiplehats** [woocommerce/woocommerce#47380] [WooCommerce, Blocks, and Site Editing]
Is it possible to fill a slot without using React? I like to use vanilla TS or Svelte.

https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/available-slot-fills.md

I guess I could do something along the lines of: 

https://youtu.be/FrusJNycQvk?si=dj4mBsnfvxLOLwvo

**Comment by senadir** [comment]
Slot Fills can only be used with React, you don't need a build system thou, but it still will be a react component, because it uses React Portals to insert that component into the existing React tree, thou nothing preventing you from having a React component orchestrate a Svelte tree.

**Reply by multiplehats** [reply]
Thanks for the quick reply! I figured that was the case, will have some figuring out to do to see if I can get it to work.

One more question, these slots do not need Gutenberg support right? I did some research a while ago, and support for other frameworks is not supported yet: https://github.com/WordPress/gutenberg/discussions/38224

**Reply by senadir** [reply]
I'm not sure what do you mean by Gutenberg support?

**Reply by multiplehats** [reply]
Let me clarify, should that component be able to (visually) render in the Gutenberg editor? No right?

**Reply by senadir** [reply]
Well ideally it should, to give merchant a correct view of their Checkout. If it works in the frontend, it should work in the editor, as we use the same Slot fill APIs.
